test/webrick/test_filehandler.rb

```diff
@@ -287,6 +288,18 @@ def test_short_filename
     end
   end
 
+  def test_cjk_in_path
+    Dir.mktmpdir("\u3042") do |dir|
+      File.write("#{dir}/\u3042.txt", "test_cjk_in_path")
+      config = { :DocumentRoot => dir }
+      TestWEBrick.start_httpserver(config) do |server, addr, port, log|
+        http = Net::HTTP.new(addr, port)
+        req = Net::HTTP::Get.new("/%E3%81%82.txt")
+        http.request(req){|res| assert_equal("200", res.code, log.call) }
+      end
+    end
+  end
+
```

Comment:
require 'webrick'

d="\u3042"
Dir.mkdir(d) if not File.exist? d
port = 8080
config = { :DocumentRoot => "." , :Port => port }
s=WEBrick::HTTPServer.new(config)
Thread.new {s.start }
url = "http://127.0.0.1:#{port}/%E3%81%82/" # from firefox
`curl #{url} `
sleep 20